test/SpokePool.Deposit.ts

```diff
@@ -402,23 +411,313 @@ describe("SpokePool Depositor Logic", async function () {
   });
 
   describe("deposit USS", function () {
+    let relayData: USSRelayData, depositArgs: any[];
+    function getDepositArgsFromRelayData(_relayData: USSRelayData, _quoteTimestamp = quoteTimestamp) {
+      return [
+        _relayData.depositor,
+        _relayData.recipient,
+        _relayData.inputToken,
+        _relayData.outputToken,
+        _relayData.inputAmount,
+        _relayData.outputAmount,
+        _relayData.destinationChainId,
+        _relayData.exclusiveRelayer,
+        _quoteTimestamp,
+        _relayData.fillDeadline,
+        _relayData.exclusivityDeadline,
+        _relayData.message,
+      ];
+    }
+    beforeEach(async function () {
+      relayData = {
+        depositor: depositor.address,
+        recipient: recipient.address,
+        exclusiveRelayer: ZERO_ADDRESS,
+        inputToken: erc20.address,
+        outputToken: randomAddress(),
+        inputAmount: amountToDeposit,
+        outputAmount: amountToDeposit.sub(19),
+        originChainId: originChainId,
+        destinationChainId: destinationChainId,
+        depositId: 0,
+        fillDeadline: quoteTimestamp + 1000,
+        exclusivityDeadline: 0,
+        message: "0x",
+      };
+      depositArgs = getDepositArgsFromRelayData(relayData);
+    });
     it("placeholder: gas test", async function () {
-      await spokePool.depositUSS(
+      await spokePool.connect(depositor).depositUSS(...depositArgs);
+    });
+    it("route disabled", async function () {
+      // Verify that routes are disabled by default for a new route
+      const _depositArgs = getDepositArgsFromRelayData({ ...relayData, destinationChainId: 999 });
+      await expect(spokePool.connect(depositor).depositUSS(..._depositArgs)).to.be.revertedWith("DisabledRoute");
+
+      // Enable the route:
+      await spokePool.connect(depositor).setEnableRoute(erc20.address, 999, true);
+      await expect(spokePool.connect(depositor).depositUSS(..._depositArgs)).to.not.be.reverted;
+    });
+    it("invalid quoteTimestamp", async function () {
+      const quoteTimeBuffer = await spokePool.depositQuoteTimeBuffer();
+      const currentTime = await spokePool.getCurrentTime();
+
+      await expect(
+        spokePool.connect(depositor).depositUSS(
+          // quoteTimestamp too far into past (i.e. beyond the buffer)
+          ...getDepositArgsFromRelayData(relayData, currentTime.sub(quoteTimeBuffer).sub(100))
```

Comment:
Should we test the absolute margin, rather than margin - 100 ?

```suggestion
          ...getDepositArgsFromRelayData(relayData, currentTime.sub(quoteTimeBuffer).sub(1))
```